deku-p/src/core/protocol/protocol.ml

```diff
@@ -73,7 +73,23 @@ let apply_operation ~current_level protocol operation :
             | Ok ledger -> (ledger, Some receipt, vm_state, None)
             | Error error -> (ledger, Some receipt, vm_state, Some error))
         | Operation_vm_transaction { sender; operation } -> (
-            let receipt = Receipt.Vm_transaction_receipt { operation = hash } in
+            let open Deku_ledger in
+            let contract_address =
+              match operation.operation with
+              | View { address; _ } -> Some address
+              | Originate _ ->
+                  Some
+                    (Contract_address.of_user_operation_hash
+                    @@ Operation_hash.to_blake2b hash)
+              | Call { address; _ } -> (
+                  match address with
+                  | Address.Implicit _ -> None
```

Comment:
This case is the reason I defined:
```ocaml
    Vm_transaction_receipt of {
      operation : Operation_hash.t;
      contract_address : Deku_ledger.Contract_address.t option;
    }
```
I am not totally convinced this is the best way to do it, but don't have anything else in mind for now...

Reply:
@d4hines I quickly exchanged with @Zett98 about that, and inside `Env.ml` something similar is done. Hence, I just did the same.
What do yo think? Did I misunderstand something?